src/openvslam/system.cc

```diff
@@ -286,15 +296,42 @@ Mat44_t system::feed_RGBD_frame(const cv::Mat& rgb_img, const cv::Mat& depthmap,
     return cam_pose_cw;
 }
 
+bool system::update_pose(const Mat44_t& pose) {
+    std::lock_guard<std::mutex> lock(mtx_tracking_);
+
+    if (!tracker_->tracking_started_) {
+        // We can not process until tracking starts.
+        // Otherwise this may cause crashes in frame_publisher_.
+        spdlog::warn("Can not update the pose: tracking was not started");
+        return false;
+    }
+
+    const bool status = tracker_->track_pose(pose);
```

Comment:
How about creating a request (like `reset_is_requested_`) instead of processing relocalization here?
I am thinking of processing it near the following line:

https://github.com/OpenVSLAM-Community/openvslam/blob/b28a597e55b35b4901e4d5cc6de83f510fa491b0/src/openvslam/tracking_module.cc#L327

The current implementation has performance problems because it does not process the following:

https://github.com/OpenVSLAM-Community/openvslam/blob/b28a597e55b35b4901e4d5cc6de83f510fa491b0/src/openvslam/tracking_module.cc#L235-L244

https://github.com/OpenVSLAM-Community/openvslam/blob/b28a597e55b35b4901e4d5cc6de83f510fa491b0/src/openvslam/tracking_module.cc#L266-L276

Reply:
Done via request_* API: https://github.com/OpenVSLAM-Community/openvslam/pull/79/files#diff-9d7c6e92591d28cb8c87b3973729fddb3abd564e9feae8861851fd48dd31b2caR188